docs/proposals/images/kubelet-authentication/client-authenticator-flow.plantuml

```diff
@@ -0,0 +1,19 @@
+@startuml client-authenticator-flow
+
+(*) --> if "client certificate" then
+  -->[file exists] if "certificate expires" then
+   -->[in less than 10 days] "Begin kubelet client CSR process"
```

Comment:
what motivates the decisions to proceed with rotation 10 days before expiration and would this and the validity period itself be configurable?

the kubelet certificate manager (that is based on client-go) uses something like "80% of the period has passed" to trigger rotation and the default validity is for 1 year.

the likelihood of users wanting this to be configurable is relatively high.

(similar Q for the serving cert flow).

Reply:
I copied the value from the current GCP, no other motivation.

Reply:
Could make it configurable, for sure.

Reply:
also iirc we only support BOY CA

Reply:
> also iirc we only support BOY CA for now

That's a totally different topic

Reply:
> the likelihood of users wanting this to be configurable is relatively high, because normally if they provide the certificates out of band with their own means they have control, and if auto-rotation is disabled.

what I'm sasying is that IIRC you can't provide your own certs in CAPI, so wanting it to be configurable makes sense only we support that

Reply:
This isn't about providing your own certificates - this is about the rotation loop of the client authenticator. We will default to 10 days, but a user might want a shorter duration.

Reply:
I think switching to 80% of the period has passed makes sense, and then we can make the controller configurable.

Reply:
Updated